src/Raven.Server/Documents/Revisions/RevisionsStorage.cs

```diff
@@ -727,13 +729,30 @@ public IEnumerable<string> GetCollections(Transaction transaction)
             return deletedRevisionsCount;
         }
 
-        public void DeleteRevision(DocumentsOperationContext context, Slice key, string collection, string changeVector, long lastModifiedTicks)
+        internal static unsafe ByteStringContext.ExternalScope CreateRevisionTombstoneKeySlice(DocumentsOperationContext context, ByteString buffer, Slice documentIdPrefix, Slice changeVectorSlice, out Slice keySlice)
+        {
+            var scope = Slice.External(context.Allocator, buffer.Ptr, buffer.Length, out keySlice);
+            documentIdPrefix.CopyTo(buffer.Ptr);
+            int pos = documentIdPrefix.Size;
+            buffer.Ptr[pos - 1] = SpecialChars.RecordSeparator;
+            changeVectorSlice.CopyTo(buffer.Ptr + pos);
+            return scope;
+        }
+
+        public void DeleteRevision(DocumentsOperationContext context, Slice key, string collection, string changeVector, long lastModifiedTicks, LazyStringValue revisionTombstoneKey = null)
         {
             var collectionName = _documentsStorage.ExtractCollectionName(context, collection);
             var table = EnsureRevisionTableCreated(context.Transaction.InnerTransaction, collectionName);
 
             long revisionEtag;
-            if (table.ReadByKey(key, out TableValueReader tvr))
+
+            Slice changeVectorSlice = key;
+            if (revisionTombstoneKey != null)
+            {
+                using var scope = RevisionTombstoneReplicationItem.TryExtractChangeVectorSliceFromKey(context.Allocator, revisionTombstoneKey, out changeVectorSlice);
```

Comment:
you dispose it immediately here, that can be a problem, no?

Reply:
Done